docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
+type errors, if the program is not correct or if the static type annotations
+aren't able to fully represent the runtime types. Removing static type
+annotations (making the program more dynamic) should not result in additional
+static type errors. This is often referred to as the **gradual guarantee**.
+
+In Python's type system, we don't take the gradual guarantee as a strict
+requirement, but it's a useful guideline.
+
+Subtype, supertype, and type equivalence
+----------------------------------------
+
+A fully static type ``B`` is a **subtype** of another fully static type ``A``
+if and only if the set of values represented by ``B`` is a subset of the set of
+values represented by ``A``. Because the subset relation on sets is transitive
+and reflexive, the subtype relation is also transitive (if ``C`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``, then ``C`` is a subtype of ``A``) and
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if and only if ``B`` is a subtype of ``A``; or equivalently, if and only
+if the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on fully static types: the types
+``A`` and ``B`` are equivalent (or "the same type") if and only if ``A`` is a
+subtype of ``B`` and ``B`` is a subtype of ``A``. This means that the set of
+values represented by ``A`` is both a superset and a subset of the values
+represented by ``B``, meaning ``A`` and ``B`` must represent the same set of
+values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` (or as a "proper
+subtype" of ``A``) if ``B`` is a subtype of ``A`` and ``B`` is not equivalent
+to ``A``. In the same scenario we can describe the type ``A`` as "wider" than
+``B``, or a "proper supertype" of ``B``.
+
+Materialization
+---------------
+
+Since :ref:`Any` represents an unknown static type, it does not represent any
+known single set of values (it represents an unknown set of values.) Thus it is
+not in the domain of the subtype, supertype, or equivalence relations on static
+types described above.
+
+To relate gradual types more generally, we define a **materialization**
+relation. The intuition for materialization is that it transforms a "more
+dynamic" type to a "more static" type. Given a gradual type ``A``, if we
+replace zero or more occurrences of ``Any`` in ``A`` with some gradual type
+(which can be different for each occurrence of ``Any``), the resulting gradual
+type ``B`` is a materialization of ``A``.
+
+For instance, ``tuple[int, str]`` (a fully static type) and ``tuple[Any, str]``
+(a gradual type) are both materializations of ``tuple[Any, Any]``. ``tuple[int,
+str]`` is also a materialization of ``tuple[Any, str]``.
+
+If ``B`` is a materialization of ``A``, we can say that ``B`` is a "more
```

Comment:
Hmm. The "we can say" construction is fairly common in presentations of formal definitions like this: it clearly indicates "heads up, here we are about to present and define a new term, with a precise meaning." I feel like the edit here doesn't communicate that as well.

Reply:
I see your point. IMHO this whole document can only be understood as a series of assertions that define otherwise unknown concepts, however. In that context, I don't feel like the extra verbosity really adds anything.

Reply:
Fair enough, but in this case I prefer to leave it as is.